src/textual/_on.py

```diff
@@ -13,47 +14,76 @@ class OnDecoratorError(Exception):
     """Errors related to the `on` decorator.
 
     Typically raised at import time as an early warning system.
-
     """
 
 
+class OnNoWidget(Exception):
+    """A selector was applied to an attribute that isn't a widget."""
+
+
 def on(
-    message_type: type[Message], selector: str | None = None
+    message_type: type[Message], selector: str | None = None, **kwargs: str
 ) -> Callable[[DecoratedType], DecoratedType]:
-    """Decorator to declare method is a message handler.
+    """Decorator to declare that the method is a message handler.
+
+    The decorator can take a CSS selector that is applied to the attribute `control`
```

Comment:
```suggestion
    The decorator accepts an optional CSS selector that is matched against a widget exposed by a `control` attribute on the message.
```

It's not "applied to" ...